src/message-stream.ts

```diff
@@ -276,10 +276,11 @@ export class MessageStream extends PassThrough {
 
     this._streams.set(stream, true);
 
-    if (!isStreamEnded(stream)) {
-      stream.once('end', () => this._onEnd(stream, status));
-    } else {
+    if (isStreamEnded(stream)) {
       this._onEnd(stream, status);
+    } else {
+      stream.once('end', () => this._onEnd(stream, status));
+      stream.push(null);
```

Comment:
Should this null push be in the once('end') handler? Or maybe I'm missing what's going on here.

Reply:
I think this is a side effect of using the native stream module, but for whatever reason in Node@6 this will hang indefinitely without it. If we ever make it to this block of code it would be because

* `status` event has been fired (no more data will be sent down)
* presumably we still have messages within the internal buffer waiting to be released

So pushing null should be safe I think.

Reply:
Countdown to the april deprecation of Node.js 6 commences 🕥